### PR TITLE
Feature/donate-command

### LINKED ISSUE #142: Donate command
We need a command to display information about donating. Would display the Patreon and possibly the cascade email to directly send money to.

### PR DESCRIPTION
**Pull request by JamesHawkinss**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Added/Changed
Added Donate command. Changed some other stuff in the bot, canny remember exactly what but a few small things.

Closes #142

### CODE REVIEW

File: src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java [hunk 1]
```diff
@@ -17,8 +17,10 @@
     public void onCommand(Member sender, CommandContext context) {
         if (Environment.isProduction()) {
             context.getDirectMessageMessaging().replyDM(CascadeBot.getInvite(), true);
+            context.getTypedMessaging().replyInfo("We've sent you a DM with more information!");
```
Comment (TortleWortle) [inline src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java]: Shouldn't this be localized?

File: src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java [hunk 2]
```diff
@@ -17,8 +17,10 @@
...
         } else {
             context.getDirectMessageMessaging().replyDM("https://www.youtube.com/watch?v=ARJ8cAGm6JE");
+            context.getTypedMessaging().replyInfo("We've sent you a DM with more information!");
```
Comment (TortleWortle) [inline src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java]: Shouldn't this be localized?

Also could just put this line after the if else.

File: src/main/java/org/cascadebot/cascadebot/CascadeBot.java [hunk 3]
```diff
@@ -220,7 +220,7 @@ private void init() {
         Thread.setDefaultUncaughtExceptionHandler(((t, e) -> LOGGER.error("Uncaught exception in thread " + t, MDCException.from(e))));
         Thread.currentThread()
                 .setUncaughtExceptionHandler(((t, e) -> LOGGER.error("Uncaught exception in thread " + t, MDCException.from(e))));
-
+
```
Comment (TortleWortle) [inline src/main/java/org/cascadebot/cascadebot/CascadeBot.java]: wat

Comment (JamesHawkinss) [reply src/main/java/org/cascadebot/cascadebot/CascadeBot.java]: ignore that, me and binary were testing something

File: src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java [hunk 4]
```diff
@@ -17,8 +17,10 @@
     public void onCommand(Member sender, CommandContext context) {
         if (Environment.isProduction()) {
             context.getDirectMessageMessaging().replyDM(CascadeBot.getInvite(), true);
+            context.getTypedMessaging().replyInfo(context.i18n("commands.invite.sent_more_info"));
         } else {
             context.getDirectMessageMessaging().replyDM("https://www.youtube.com/watch?v=ARJ8cAGm6JE");
+            context.getTypedMessaging().replyInfo(context.i18n("commands.invite.sent_more_info"));
```
Comment (TortleWortle) [inline src/main/java/org/cascadebot/cascadebot/commands/core/InviteCommand.java]: Could remove line 20 and move this one line down.

**Comment by weeryan17** [comment]
I think this pr should wit for me new donation systems to be finished as I think it would be a good idea to also have this command show what people will be getting for donating instead of just giving list as just pointing people at links doesn't make them want to click them and actually give us money.

**Comment by JamesHawkinss** [comment]
Do you know when we'll have all that decided?

**Comment by weeryan17** [comment]
> 
> 
> Do you know when we'll have all that decided?

the donation teirs? they're already decided, and are in the community planning channel on the discord. Although i don't want us to hard code the tiers, and I'm currently working on a donation system that will be more expandable, and give us the ability to implement new tiers more easily.

**Comment by JamesHawkinss** [comment]
Is this PR going ahead?

**Comment by mrjvs** [comment]
> 
> 
> Is this PR going ahead?

we're still working on the donation tier system. when that is finished, this can continue

**Comment by weeryan17** [comment]
@JamesHawkinss this can go ahead now. the new system is now done. when you re-make this pr pr it to the donation-stuffs branch.

**Pull request closed without merging**